app/queries/answers.ts

```diff
@@ -19,43 +19,24 @@ export async function getQuestionsNeedingCorrectAnswer() {
       revealAtAnswerCount: number;
     }[]
   >`
-    SELECT
-      "id",
-      "answerCount",
-      "revealAtDate",
-      "revealAtAnswerCount"
-    FROM (
-      SELECT
-        q.id,
-        "revealAtDate",
-        "revealAtAnswerCount",
-        (
-          SELECT
-            COUNT(DISTINCT CONCAT(qa."userId", qo."questionId"))
-          FROM
-            "QuestionOption" qo
-          JOIN
-            "QuestionAnswer" qa ON qa."questionOptionId" = qo."id"
-          WHERE
-            qo."questionId" = q."id"
-      ) AS "answerCount"
-      FROM
-        "Question" q
-      LEFT JOIN
-        "QuestionOption" qo ON q.id = qo."questionId"
-      LEFT JOIN
-        "QuestionAnswer" qa ON qo.id = qa."questionOptionId"
-      WHERE
-        ("calculatedIsCorrect" IS NULL OR "calculatedAveragePercentage" IS NULL)
-      GROUP BY q.id
-    )
-    WHERE (
-      (
-        ("revealAtDate" IS NOT NULL AND "revealAtDate" < NOW())
-        OR ("revealAtAnswerCount" IS NOT NULL AND "revealAtAnswerCount" <= "answerCount")
-      ) AND (
-        "answerCount" >= ${Number(process.env.MINIMAL_ANSWERS_PER_QUESTION)}
+      WITH AnswerCounts AS (
+          SELECT qo."questionId",
+                 COUNT(DISTINCT qa."userId") AS "answerCount"
+          FROM "QuestionOption" qo
+                   JOIN "QuestionAnswer" qa ON qa."questionOptionId" = qo."id"
+          WHERE qo."calculatedIsCorrect" IS NULL OR qo."calculatedAveragePercentage" IS NULL
+          GROUP BY qo."questionId"
       )
-    )
+      SELECT q.id,
+             q."revealAtDate",
+             q."revealAtAnswerCount",
+             sub."answerCount"
+      FROM "Question" q
+               JOIN AnswerCounts sub ON sub."questionId" = q.id
+      WHERE
+          (q."revealAtDate" IS NOT NULL AND q."revealAtDate" < NOW())
+         OR
+          ((q."revealAtAnswerCount" IS NOT NULL AND q."revealAtAnswerCount" <= sub."answerCount")
+              AND sub."answerCount" >= ${Number(process.env.MINIMAL_ANSWERS_PER_QUESTION)});
```

Comment:
I think that where condition logic should be same.
```suggestion
      WHERE (
          (
              (q."revealAtDate" IS NOT NULL AND q."revealAtDate" < NOW())
              OR (q."revealAtAnswerCount" IS NOT NULL AND q."revealAtAnswerCount" <= sub."answerCount")
          ) AND (
              sub."answerCount" >= ${Number(process.env.MINIMAL_ANSWERS_PER_QUESTION)}
          )
      );
```